File: Components/DevExpressVCL/Demos/ExpressRichEdit Control/CBuilder/Common/EBarsUtils.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#include <Forms.hpp>
#pragma hdrstop

#include "EBarsUtils.h"
#include "EBarsDemoRating.h"
#include "AboutDemoForm.h"
#pragma package(smart_init)
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TdmCommonData *dmCommonData;
//---------------------------------------------------------------------------
__fastcall TdmCommonData::TdmCommonData(TComponent* Owner)
        : TDataModule(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TdmCommonData::actSBarsHelpExecute(TObject *Sender)
{
  Application->HelpFile = "..\\..\\..\\..\\ExpressSideBar\\Help\\eSideBar.hlp";
  Application->HelpCommand(HELP_FINDER, 0);
}
//---------------------------------------------------------------------------
void __fastcall TdmCommonData::actBarsHelpExecute(TObject *Sender)
{
  Application->HelpFile = "..\\..\\..\\Help\\ExpressBars.hlp";
  Application->HelpCommand(HELP_FINDER, 0);
}
//---------------------------------------------------------------------------
void __fastcall TdmCommonData::actDockingHelpExecute(TObject *Sender)
{
  Application->HelpFile = "..\\..\\..\\..\\ExpressDocking Library\\Help\\docking.hlp";
  Application->HelpCommand(HELP_FINDER, 0);
}
//---------------------------------------------------------------------------
void __fastcall TdmCommonData::actRateDemoExecute(TObject *Sender)
{
  TEBarsDemoRatingForm* EBarsDemoRatingForm = new TEBarsDemoRatingForm(NULL);
  try {
    EBarsDemoRatingForm->ShowModal();
  }
  __finally {
    delete EBarsDemoRatingForm;
  }
}
//---------------------------------------------------------------------------
void __fastcall TdmCommonData::actMyDXExecute(TObject *Sender)
{
  Browse(spMyDX);
}
//---------------------------------------------------------------------------
void __fastcall TdmCommonData::actDownloadsExecute(TObject *Sender)
{
  Browse(spDownloads);
}
//---------------------------------------------------------------------------
void __fastcall TdmCommonData::actSupportExecute(TObject *Sender)
{
  Browse(spSupport);
}
//---------------------------------------------------------------------------
void __fastcall TdmCommonData::actDXOnTheWebExecute(TObject *Sender)
{
  Browse(spStart);
}
//---------------------------------------------------------------------------
void __fastcall TdmCommonData::actProductsExecute(TObject *Sender)
{
  Browse(spProducts);
}
//---------------------------------------------------------------------------
void __fastcall TdmCommonData::actAboutExecute(TObject *Sender)
{
  ShowAboutDemoForm();
}
//---------------------------------------------------------------------------
void Browse(dxSitePage ASitePage)
{
  PCHAR AURL;
  switch (ASitePage)
  {
    case spDownloads: AURL = dxDownloadURL; break;
    case spSupport: AURL = dxSupportURL; break;
    case spStart: AURL = dxStartURL; break;
    case spProducts: AURL = dxProductsURL; break;
    case spMyDX: AURL = dxMyDXURL; break;
  }
  ShellExecute(0, "OPEN", AURL, NULL, NULL, SW_SHOWMAXIMIZED);
}
//---------------------------------------------------------------------------

File: Components/DevExpressVCL/Demos/ExpressBars/Delphi/Common/EBarsUtils.pas
unit EBarsUtils;

interface

uses
  SysUtils, Classes, Forms, ActnList, ImgList, Controls;

const
  dxDownloadURL = 'http://www.devexpress.com/downloads';
  dxSupportURL = 'http://www.devexpress.com/Support/Center';
  dxStartURL = 'http://www.devexpress.com';
  dxProductsURL = 'http://www.devexpress.com/products';
  dxMyDXURL = 'https://www.devexpress.com/ClientCenter';

type
  dxSitePage = (spDownloads, spSupport, spStart, spProducts, spMyDX);

  TdmCommonData = class(TDataModule)
    alMain: TActionList;
    actDockingHelp: TAction;
    actRateDemo: TAction;
    actDownloads: TAction;
    actSupport: TAction;
    actDXOnTheWeb: TAction;
    actProducts: TAction;
    actAbout: TAction;
    actBarsHelp: TAction;
    actMyDX: TAction;
    actSBarsHelp: TAction;
    procedure actSBarsHelpExecute(Sender: TObject);
    procedure actDockingHelpExecute(Sender: TObject);
    procedure actBarsHelpExecute(Sender: TObject);
    procedure actRateDemoExecute(Sender: TObject);
    procedure actMyDXExecute(Sender: TObject);
    procedure actDownloadsExecute(Sender: TObject);
    procedure actDXOnTheWebExecute(Sender: TObject);
    procedure actSupportExecute(Sender: TObject);
    procedure actProductsExecute(Sender: TObject);
    procedure actAboutExecute(Sender: TObject);
  end;

procedure Browse(ASitePage: dxSitePage);

var
  dmCommonData: TdmCommonData;

implementation

{$R *.dfm}

uses
  Windows, ShellAPI, EBarsDemoRating, AboutDemoForm;

procedure Browse(ASitePage: dxSitePage);
var
  AURL: string;
begin
  case ASitePage of
    spDownloads: AURL := dxDownloadURL;
    spSupport: AURL := dxSupportURL;
    spStart: AURL := dxStartURL;
    spProducts: AURL := dxProductsURL;
    spMyDX: AURL := dxMyDXURL;
  end;
  ShellExecute(0, 'OPEN', PChar(AURL), nil, nil, SW_SHOW);
end;

procedure TdmCommonData.actSBarsHelpExecute(Sender: TObject);
begin
  Application.HelpFile := '..\..\..\..\ExpressSideBar\Help\eSideBar.hlp';
  Application.HelpCommand(HELP_FINDER, 0);
end;

procedure TdmCommonData.actDockingHelpExecute(Sender: TObject);
begin
  Application.HelpFile := '..\..\..\..\ExpressDocking Library\Help\docking.hlp';
  Application.HelpCommand(HELP_FINDER, 0);
end;

procedure TdmCommonData.actBarsHelpExecute(Sender: TObject);
begin
  Application.HelpFile := '..\..\..\Help\ExpressBars.hlp';
  Application.HelpCommand(HELP_FINDER, 0);
end;

procedure TdmCommonData.actRateDemoExecute(Sender: TObject);
begin
  with TEBarsDemoRatingForm.Create(Self) do
    try
      ShowModal;
    finally
      Free;
    end;
end;

procedure TdmCommonData.actMyDXExecute(Sender: TObject);
begin
  Browse(spMyDX);
end;

procedure TdmCommonData.actDownloadsExecute(Sender: TObject);
begin
  Browse(spDownloads);
end;

procedure TdmCommonData.actDXOnTheWebExecute(Sender: TObject);
begin
  Browse(spStart);
end;

procedure TdmCommonData.actSupportExecute(Sender: TObject);
begin
  Browse(spSupport);
end;

procedure TdmCommonData.actProductsExecute(Sender: TObject);
begin
  Browse(spProducts);
end;

procedure TdmCommonData.actAboutExecute(Sender: TObject);
begin
  ShowAboutDemoForm;
end;

end.
